test/lib/sstable_test_env.hh

```diff
@@ -101,7 +101,7 @@ struct test_env_config {
     size_t available_memory = memory::stats().total_memory();
 };
 
-data_dictionary::storage_options make_test_object_storage_options();
+data_dictionary::storage_options make_test_object_storage_options(std::optional<std::string> type = {});
```

Comment:
The fact that it "defaults to S3" is not obvious. I mean -- it obvious to those who is with this code for a long time, because there's nothing else other than S3 up until now, but I'd expect this argument to be non-optional std::string (maybe string_view) without defaults and existing callers updated to explicitly put "S3" here